cli/npm/resolution.rs

```diff
@@ -257,9 +264,23 @@ async fn add_package_reqs_to_snapshot(
     return Ok(snapshot); // already up to date
   }
 
-  let result = snapshot.resolve_pending(package_reqs, api).await;
+  let result = snapshot.resolve_pending(package_reqs.clone(), api).await;
   api.clear_memory_cache();
-  let snapshot = result?; // propagate the error after clearing the memory cache
+  let snapshot = match result {
+    Ok(snapshot) => snapshot,
+    Err(NpmResolutionError::Resolution(err)) if api.mark_force_reload() => {
+      log::debug!("{err:#}");
+      log::debug!("npm resolution failed. Trying again...");
+
+      // try again
+      let snapshot = get_new_snapshot();
+      let result = snapshot.resolve_pending(package_reqs, api).await;
+      api.clear_memory_cache();
+      // now surface the result after clearing the cache
+      result?
```

Comment:
One thing I should have done is just surface the original error here. I didn't think about that until now. I will do that in a future PR.

Reply:
Actually, this might be better.